x509/helper.go

```diff
@@ -23,8 +23,34 @@ import (
 	"github.com/notaryproject/notation-core-go/signature"
 )
 
-func isSelfSigned(cert *x509.Certificate) (bool, error) {
-	return isIssuedBy(cert, cert)
+// validateSelfSignedCert validates a self-signed certificate.
+//
+// Parameters:
+//   - cert: The certificate to validate.
+//   - isSingleCertChain: A boolean value indicating whether the certificate is
+//     the only certificate in the chain.
+//     1. is true: The certificate is both a self-signed leaf and root
+//     certificate.
+//     2. is false: The certificate is a self-signed root CA certificate.
+//
+// Note: A self-signed certificate must have the same subject and issuer.
+func validateSelfSignedCert(cert *x509.Certificate, isSingleCertChain bool) error {
```

Comment:
On a second thought, do we even need to change method `isSelfSigned`? How about keeping the contents in `helper.go` as is. Because the issue only happens in `cert chain length eq 1` scenario.

Reply:
So the original design is fine, and it's just a naming issue with the `validateSelfSignedLeaf` function?